agent/connect/ca/plugin/provider.proto

```diff
@@ -30,9 +30,10 @@ service CA {
 
 message ConfigureRequest {
     string cluster_id = 1;
-    bool is_root = 2;
-    bytes config = 3; // JSON-encoded structure
-    bytes state = 4; // JSON-encoded structure
+    string datacenter = 2;
+    bool primary_dc = 3;
+    bytes config = 4; // JSON-encoded structure
+    bytes state = 5; // JSON-encoded structure
```

Comment:
I know you are not supposed to renumber the fields. It's Protobuf cardinal sin #1. But we aren't actually using this interface for anything yet - it's not plumbed in so we might as well start with it in a clean state when we do.

Reply:
Maybe it's worth having a big comment in this file at the top mentioning that it's technically dead code for now and not to worry about this sort of stuff yet?

Reply:
I went one better and deleted this whole package as we are very likely to make lots more changes before we actually export plugins properly.